internal/api/main_api.go

```diff
@@ -264,11 +266,21 @@ func Start(ctx context.Context, node host.Host) {
 
 	// Create a new ServeMux to combine the API handler and metrics handler
 	mux := http.NewServeMux()
+
+	// Health check endpoint for service monitoring
+	mux.HandleFunc("/api/health", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`{"status":"ok"}`))
+	})
+
 	mux.Handle("/", apiHandler)
 	mux.Handle("/metrics", promhttp.Handler())
 
-	// Check if port 8080 is in use. If so, increment until an open port is found.
-	var port int = 9090
+	// Get port from PORT environment variable, default to 9090
+	port := getAPIPort()
+
+	// Check if port is in use. If so, increment until an open port is found.
 	for {
 		addr := fmt.Sprintf("0.0.0.0:%d", port)
```

Comment:
Let's add a comment here similar to the other one regarding the new issue related to adding support for selecting the interfaces and IP versions.